src/Common/LRUCache.h

```diff
@@ -57,23 +76,38 @@ class LRUCache
         return res;
     }
 
-    void set(const Key & key, const MappedPtr & mapped)
+    /**
+     * set() will fail (return false) if there is no space left and no keys could be evicted.
+     * Eviction permission of each key is defined by EvictPolicy. In default policy there is no restriction.
+     */
+    bool set(const Key & key, const MappedPtr & mapped)
     {
         std::lock_guard lock(mutex);
 
-        setImpl(key, mapped, lock);
+        return setImpl(key, mapped, lock);
+    }
+
+    /// Returns std::pair of the cached value and a bool indicating whether the value was produced during this call.
+    template <typename LoadFunc>
+    std::pair<MappedPtr, bool> getOrSet(const Key & key, LoadFunc && load_func)
+    {
+        auto [value, is_loaded, _] = getOrTrySet(key, std::move(load_func));
+        return std::make_pair(value, is_loaded);
     }
 
     /// If the value for the key is in the cache, returns it. If it is not, calls load_func() to
     /// produce it, saves the result in the cache and returns it.
-    /// Only one of several concurrent threads calling getOrSet() will call load_func(),
+    /// Only one of several concurrent threads calling getOrTrySet() will call load_func(),
     /// others will wait for that call to complete and will use its result (this helps prevent cache stampede).
     /// Exceptions occurring in load_func will be propagated to the caller. Another thread from the
     /// set of concurrent threads will then try to call its load_func etc.
     ///
-    /// Returns std::pair of the cached value and a bool indicating whether the value was produced during this call.
+    /// return std::tuple is <MappedPtr, is_value_loaded, is_value_updated>, where
+    ///  - is_value_loaded indicates whether the value was produce during this call
+    ///  - is_value_updated indicates whether the value is updated in the cache when is_value_loaded = true.
+    ///    if is_value_loaded = false, is_value_updated = false
     template <typename LoadFunc>
-    std::pair<MappedPtr, bool> getOrSet(const Key & key, LoadFunc && load_func)
+    std::tuple<MappedPtr, bool, bool> getOrTrySet(const Key &key, LoadFunc && load_func)
```

Comment:
I did not find where you use getOrTrySet apart from in getOrSet() method, may be I missed the place, can you say?

Reply:
This method is added for. 
> But what if we care for the current call, not for next. So we call getOrSet and do not care that set succeded? May be after all it makes sense to move this logic out as suggested here

Reply:
Ok, I see.